### ISSUE TITLE
primary shard of stable index become unassigned when other indices trigger TooManyOpenFiles

### ISSUE THREAD

**Issue by sweetest** [elastic/elasticsearch#15134]
in ES version 1.7.3, when cluster is green and another index is created, then triggres too many open files error, some primary shards of online indices which were stable become unassigned, and then those shards become disabled and cluster reroute API also does not work for those shards.

**Comment by sweetest** [comment]
And also I'd like to ask you if it's possible to recover those disabled shards without losing data.

**Comment by sweetest** [comment]
I tried to reroute that shard to where the actual data exists, with allow_primary option set to true, but resulted in ES creating a new empty shards and deleting all existing data.

POST /_cluster/reroute
{
   "commands": [
      {
         "allocate": {
            "index": "data-2015-12-01",
            "shard": 1,
            "node": "node03",
            "allow_primary": true
         }
      }
   ]
}

**Comment by sweetest** [comment]
this is the result I got from POST /_cluster/reroute?explain&pretty

```
   "explanations": [
      {
         "command": "allocate",
         "parameters": {
            "index": "data-2015-12-01",
            "shard": 1,
            "node": "node03",
            "allow_primary": false
         },
         "decisions": [
            {
               "decider": "allocate_allocation_command",
               "decision": "NO",
               "explanation": "trying to allocate a primary shard [nelo2-log-2015-07-01][3], which is disabled"
            }
         ]
      }
```

**Comment by clintongormley** [comment]
@sweetest the problem is too few file handles, which is a config option that should be set up before elasticsearch starts.  We can't work around this, it is like having a full disk.  Once you fix that problem (and restart Elasticsearch) the existing primaries should recover.  It may be that you have corrupted transaction logs which prevent recovery (this has been fixed in 2.0) but you can delete those and allow recovery to continue.
